file_upload.rb

```diff
@@ -0,0 +1,144 @@
+  checksum = Digest::MD5.base64digest(File.read(file_path))
+
+  gql = <<~GQL
+    mutation CreateFileUploadDemo {
+      create_file_upload (
+        filename: "#{filename}"
+        content_type: "#{content_type}"
+        byte_size: #{byte_size}
+        checksum: "#{checksum}"
+      ) {
+        id
+        url
+        headers
+      }
+    }
+  GQL
+  request.body = { 'query': gql }.to_json
+  response = http.request(request)
+  puts "create_file_upload response.code = #{response.code}"
+  puts "create_file_upload response.body = #{response.body}"
+
+  json = JSON.parse(response.body)
+  json.dig('data', 'create_file_upload')
+end
+
+def upload_file(url, headers, file_path)
+  #
+  # This method demonstrates how to upload a file to
+  # Sentera's cloud storage using the URL and headers
+  # that were retrieved via the create_file_upload
+  # GraphQL mutation.
+  #
+  uri = URI(url)
+  file_contents = File.read(file_path)
+  Net::HTTP.start(uri.host) do |http|
+    response = http.send_request('PUT',
+                                 uri,
+                                 file_contents,
+                                 headers)
+    puts "upload_file response.code = #{response.code}"
+  end
+end
+
+def use_file(file_id)
+  #
+  # This method demonstrates how to use the ID of a file that
+  # was previously uploaded to Sentera's cloud storage with one
+  # of the mutations in Sentera's GraphQL API that accepts a
+  # file ID as an input. In this example, we'll use the
+  # import_files GraphQL mutation, and attach the file to
+  # a feature set that the caller is permitted to access.
+  #
+  uri = URI(GQL_ENDPOINT)
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = GQL_ENDPOINT.include?('https://')
+  headers = {
+    'Content-Type': 'application/json',
+    'Authorization': "Bearer #{AUTH_TOKEN}"
+  }
+  request = Net::HTTP::Post.new(uri.path, headers)
+
+  owner_sentera_id = '<Your feature set ID goes here>'
+
+  gql = <<~GQL
+    mutation UseFileDemo {
+      import_files (
+        owner_type: FEATURE_SET
+        owner_sentera_id: "#{owner_sentera_id}"
+        file_type: FLIGHT_LOG
+        file_keys: ["#{file_id}"]
+      ) {
+        status
+      }
+    }
+  GQL
+  request.body = { 'query': gql }.to_json
+  response = http.request(request)
+  puts "use_file response.code = #{response.code}"
+  puts "use_file response.body = #{response.body}"
+
+  json = JSON.parse(response.body)
+  json.dig('data', 'import_files')
+end
+
+# -------------------------------------------------------------
+# MAIN
+
+file_path = 'example_flight_log.json'
+content_type = 'application/json'
+
+# Step 1: Create a file upload
+file_upload_json = create_file_upload(file_path, content_type)
+
+# Step 2: Upload the file
+upload_url = file_upload_json['url']
+upload_headers = file_upload_json['headers']
+upload_file(upload_url, upload_headers, file_path)
+
+# Step 3: Use the file
+file_id = file_upload_json['id']
+use_file(file_id)
+
+puts 'Done!'
```

Comment:
This is great! These `MAIN` lines make it very clear how this all works. This 3-step process is going to be identical for all clients. Do you think it'd be useful to roughly describe it (essentially just copy/paste the `# Step n` comments) in the README? Just seems like communicating in rough terms that there are three required steps would help newcomers to the library, before they read any code.

Reply:
I suppose this snippet from your intro comment could be a candidate to move to the README as well:
```
# Full documentation of this workflow can be found here:
# https://api.sentera.com/api/getting_started/uploading_files.html
```

Reply:
Done: https://github.com/SenteraLLC/file-upload-examples/pull/1/commits/48d8d5828397c65df05104955330f30997e6893c

![file-upload-examples-readme](https://user-images.githubusercontent.com/123218/98830349-ec1eb480-23ff-11eb-8c38-612e6d35fdac.png)